src/testapi_matrix_grdfill.c

```diff
@@ -0,0 +1,25 @@
+#include "gmt.h"
+
+/* Test calling grdfill on a grid with an asymmetric pad
+ */
+
+int main () {
+	unsigned int mode = GMT_SESSION_EXTERNAL;
+	struct GMT_MATRIX *M_g = NULL;
+	char input_g[GMT_VF_LEN] = {""};
+	char args_g[128]= {""};
+	struct GMTAPI_CTRL *API = NULL;
+
+	API = GMT_Create_Session ("test", 2U, mode, NULL);
```

Comment:
This regression test doesn't actually reproduce the original crash because the pad for this example is still `[2, 2, 2, 2]`. The pad for the failing PyGMT example is `[1, 3, 1, 6]`. @PaulWessel I know `GMT_Create_Session` can be used to change the default pad from `2`; is there a way to make the pad asymmetric?

Reply:
Not via Create_Session but you could run
`
`	gmt_M_grd_setpad (GMT, header, pad);	/* Copy the pad to the header */

where pad is a 4-array and those numbers can be anything.  Then, the header->pad is set and then you must do 
`	gmt_set_grddim (GMT, header);		/* Update all dimensions */`
to have all dimensions updated.  Once that is done you the ->size is set and you can allocate array.
Those calls require gmt_dev.h though.

Reply:
I tried that workflow, but the pad gets reset again before reaching the problematic section of grdfill. It may be necessary to add a regression test in the PyGMT repository instead.

Reply:
I pushed the changes to the test in https://github.com/GenericMappingTools/gmt/pull/6418/commits/271b051a64d258eb7ec24ab42db6f7aefa911f51. Do you mind taking a look to see if I am missing anything?